operator/pkg/ciliumendpointslice/manager.go

```diff
@@ -390,6 +374,40 @@ func (c *cesMgr) RemoveCEPFromCache(cepName string, baseDelay time.Duration) {
 	return
 }
 
+func (c *cesMgr) removeCEPFromCES(cepName string, cesName string, baseDelay time.Duration, identity int64, checkIdentity bool) {
+	var ces *cesTracker
+	var exists bool
+	if ces, exists = c.desiredCESs.getCESTracker(cesName); !exists {
+		log.WithFields(logrus.Fields{
+			logfields.CESName: cesName,
+			logfields.CEPName: cepName,
+		}).Info("Attempted to remove non-existent CES, skipping.")
+		return
+	}
+
+	ces.backendMutex.Lock()
+	defer ces.backendMutex.Unlock()
+	for i, ep := range ces.ces.Endpoints {
+		if GetCEPNameFromCCEP(&ep, ces.ces.Namespace) == cepName && (!checkIdentity || ep.IdentityID == identity) {
+			// Insert deleted CoreCEP in removedCEPs
+			ces.removedCEPs[GetCEPNameFromCCEP(&ep, ces.ces.Namespace)] = struct{}{}
+			ces.ces.Endpoints =
+				append(ces.ces.Endpoints[:i],
+					ces.ces.Endpoints[i+1:]...)
```

Comment:
do we expect more than a few endpoints to pass this check and be removed? if yes, this is a good candidate for optimization

Reply:
This section is moved from a RemoveCEPFromCache function.

Before this PR it would always remove 1 endpoint.
With this PR it is possible (and tested) that it can remove more than one in case something really wrong happened but I don't think it is possible now except for some bugs.

So - the point of this PR is to make sure the state is OK after operator restart regardless of anything.
In practice I don't think in the current code it is possible to have duplicated CEP in the same CES.
However it is possible that different CEPs would be removed from the same CES.

Reply:
Performance-wise, it should be insignificant, unless the duplicate list is big, but then we have another big problem.